geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/StatParser.scala

```diff
@@ -66,8 +66,17 @@ private class StatParser extends BasicParser {
     oneOrMore(singleStat, ";") ~~> { s => if (s.length == 1) s.head else new SeqStat(s) } ~ EOI
   }
 
+  def groupBy: Rule1[Stat] = rule {
+    // TODO: Clean up the DSL here
+    // TODO: Maybe support multiple stats in a groupby.
+    "GroupBy(" ~ string ~ "," ~ singleStat ~ ")" ~~> { (attribute, groupedStats) =>
```

Comment:
something that would simplify this PR:
match singleStat as a string, but validate it as a parsable stat. Then pass the stat string into GroupBy instead of the stat instance. Then instead of all the `newCopy` methods, just re-parse the stat string when you want a copy.